File: app/src/main/java/com/example/learnwordstrainer/viewmodels/MainViewModel.java
package com.example.learnwordstrainer.viewmodels;

import android.app.Application;
import androidx.appcompat.app.AppCompatDelegate;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.example.learnwordstrainer.repository.ThemeRepository;
import com.example.learnwordstrainer.repository.WordRepository;

public class MainViewModel extends AndroidViewModel {
    private final WordRepository wordRepository;
    private final ThemeRepository themeRepository;

    private final MutableLiveData<Integer> totalWordsCount = new MutableLiveData<>(0);
    private final MutableLiveData<Integer> learnedPercentage = new MutableLiveData<>(0);



    public MainViewModel(Application application) {
        super(application);
        wordRepository = new WordRepository(application);
        themeRepository = new ThemeRepository(application);

        // Застосування збереженої теми при старті
        AppCompatDelegate.setDefaultNightMode(themeRepository.getThemeMode());

        // Початкове завантаження статистики
        loadStatistics();
    }

    public LiveData<Integer> getTotalWordsCount() {
        return totalWordsCount;
    }

    public LiveData<Integer> getLearnedPercentage() {
        return learnedPercentage;
    }

    public void loadStatistics() {
        int total = wordRepository.getWordCount();
        int learned = wordRepository.getLearnedWordsCount();

        totalWordsCount.setValue(total);

        int percentage = total > 0 ? (learned * 100) / total : 0;
        learnedPercentage.setValue(percentage);
    }
}

File: app/src/main/java/com/example/learnwordstrainer/viewmodels/MainViewModel.kt
package com.example.learnwordstrainer.viewmodels

import android.app.Application
import androidx.appcompat.app.AppCompatDelegate
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.example.learnwordstrainer.repository.ThemeRepository
import com.example.learnwordstrainer.repository.WordRepository
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.asStateFlow
import javax.inject.Inject

@HiltViewModel
class MainViewModel @Inject constructor(
    private val themeRepository: ThemeRepository,
    private  val wordRepository: WordRepository) : ViewModel() {

    private val _themeMode = MutableStateFlow(themeRepository.themeMode)
    val themeMode = _themeMode.asStateFlow()

    private val _totalWordsCount = MutableLiveData(0)
    val totalWordsCount: LiveData<Int> = _totalWordsCount

    private val _learnedPercentage = MutableLiveData(0)
    val learnedPercentage: LiveData<Int> = _learnedPercentage

    init {
        loadStatistics()
    }

    fun loadStatistics() {
        val total = wordRepository.getWordCount()
        val learned = wordRepository.getLearnedWordsCount()

        _totalWordsCount.value = total
        _learnedPercentage.value = if (total > 0) (learned * 100) / total else 0
    }

    fun updateTheme() {
        _themeMode.value = themeRepository.themeMode
        AppCompatDelegate.setDefaultNightMode(themeRepository.themeMode)
    }
}